### DISCUSSION TITLE
How to add Event Listeners to the custom Nodes

### DISCUSSION THREAD

**Discussion by Riteshp2001** [xyflow/xyflow#3839] [Q&A]
I want to add three buttons inside a custom node, each performing different functionalities. One of them will delete the current node along with its corresponding edge connections. However, I am stuck on how to use the OnClick event listener for that button on node.
if possible pls guide me or any other alternative as how can i do

**Comment by moklick** [comment]
You just add a button with an `onClick` handler. It's a regular React component: https://codesandbox.io/p/sandbox/silly-forest-h695t3?file=/src/CustomNode.tsx:22,1-25,1

I would add the class name "nodrag" in order to prevent dragging when pressing the button.

**Reply by Riteshp2001** [reply]
thanks for your help really!!, you saved my day !
